students/jongbeom/users/views.py

```diff
@@ -44,7 +42,19 @@ def post(self, request):
                 phone_number = data["phone_number"],
             )
             user.save()
-            return JsonResponse({"message" : "SUCCESS"}, status=201)
-        except KeyError:
+            return JsonResponse({" message" : "SUCCESS"}, status=201)
+        except KeyError: 
             return JsonResponse({"message" : "KeyError"}, status=400)
 
+class LogInView(View):
+    def post(self, request):
+        data     = json.loads(request.body)
+        email    = data["email"]
+        password = data["password"]
+
+        try:              
+            if User.objects.filter(email=email, password=password).exists():
+                return JsonResponse({"message" : "SUCCESS"}, status=200)
+            return JsonResponse({"message" : "INVALID_USER"}, status=401)
+        except KeyError:
+            return JsonResponse({"message" : "KEY_ERROR"}, status=400)
```

Comment:
KeyError를 잡기 위한 try문의 범위가 수정되어야될 것 같습니다. 아래 코드 참고 부탁드려요~

```suggestion
        try:          
            email    = data["email"]
            password = data["password"]    
            if User.objects.filter(email=email, password=password).exists():
                return JsonResponse({"message" : "SUCCESS"}, status=200)
            return JsonResponse({"message" : "INVALID_USER"}, status=401)
        except KeyError:
            return JsonResponse({"message" : "KEY_ERROR"}, status=400)
```

Reply:
말씀해주신 것 처럼 변수가 try문 범위 때문에 수정 전 코드로 테스트시 서버에서는 다음과 같이 `KeyError: 'password' [17/Jan/2022 06:23:45] "POST /users/login HTTP/1.1" 500 68783`가 발생했다고 나오지만 httpie에서는 "message" : "KEY_ERROR" 라는 출력문이 나오지 않고 에러가 발생하였습니다. 현재는 수정 후 테스트해보니 KeyError 캐치가 제대로 되는걸 확인했습니다 감사합니다 병민님!